src/plugins/lwhpolar/README.md

```diff
@@ -0,0 +1,98 @@
+Polar Coordinate Positioning Plugin
+===================================
+
+Preface
+-------
+This plugin replaces the "rel" relative positioning plugin. 
+It includes and enhances its functionality.
+Using both will confuse the positioning, as the first one called replaces 
+relative positions by absolute ones, thus disabling the second one.
+
+Relative Positioning (Cartesian)
+--------------------------------
+The plugin uses the following attributes for a step:
+	data-rel-ref
+	data-rel-x
+	data-rel-y
+	data-rel-z
+
+If the attribute "data-rel-ref" is not given, relative coordinates will be measured
+from the position of the previous frame. If added, the step with the given id determines
+the origin. This allows to group steps to a topic without the risk of scattering them when 
+one of them is removed.
+In addition to plain numbers, which are pixel values, it is also possible to
+define relative positions as a multiple of normal step height and width, using
+a unit of "h" or "w", respectively, appended to the number.
+
+Example:
+	<div id="relstep" class="step" 
+		data-rel-ref="main_step" data-rel-x="1w" data-rel-y="0.5h" data-rel-z="0">
+			
+Relative coordinates not given are inherited from the previous step, so make sure
+to set them to zero if that behaviour is not desired.
+Absolute values always override relative values!
+
+Polar Positioning
+-----------------
+
+To place steps e.g. in a circle around a central thought, polar positioning is introduced here.
+It uses the following attributes for a step:
+	data-rel-ref
+	data-rho
+	data-phi
+	data-rel-rho
+	data-rel-phi
+
+Rho is the radius (the distance from the origin) and phi the angle in degree, counting from the 
+horizontal (positive x, "03:00hrs") counterclockwise.
+In addition to plain numbers, which are pixel values, it is also possible to define the radius 
+as a multiple of the normal step diagonal, using a unit of "d" appended to the number.
+
+Relative polar coordinates are not added up, they just count from the given origin. For absolute
+coordinates that is (0,0,0), for relative ones the origin of the last step or the one pointed to 
+by the "data-rel-ref" attribute. 
+Alternatively, a new relative origin can be set by giving absolute cartesian coordinates.
```

Comment:
This explanation still leaves open questions. (Didn't read the code yet, but as these are not discussed in documentation, it's unclear how much you've thought about these questions and possible alternatives...)

* It seems that *rho* and *phi* are only translated into `data-x` and `data-y`. What sets `data-z`?
* Following from the same point, all steps will be facing the same way, right? So they could be placed in a circle around an origin, but their upper edge will point "north" for all of them? An alternative could be for their upper edge to point toward the origin. I guess that can be achieved with using `data-rotate-z` to equal `data-polar-phi`?
* From this documentation, I expect that a step using `data-polar-ref` will inherit the `data-z` value from the references step. Correct?
* Also, from this documentation I expect that the polar coordinates are always applied in the x,y-plane. In other words, if the referenced step uses `data-rotate-x` or `data-rotate-y`, it will itself not be in the x,y-plane, but the steps positioned relative to it will? In other words, the circle drawn by *rho* and *phi* is always drawn in the x,y-plane and NOT relative to the plane defined by the referenced step?

Finally, as impress.js allows 3D positioning, wouldn't it be more appropriate to implement spherical coordinates instead of polar? This is an important question, as it affects the plugin name, which affects what all the attributes will be called.

Reply:
For what I needed (and wrote the plugin for) changing z was not required, so I kept the existing settings. The referenced steps and their children are always in the x/y-plane.

Rotation is not affected, so it can be done as You suggest. I didn't think about that. Might be an idea for the presentation...

As mentioned in the issue, I was just offering what I had already done, which is not meant to be the complete solution.
But anyway, does that really make sense? How would spherically arranged slides be orientated? It might cause spacial disorientation in the beholder ;-)

Reply:
> As mentioned in the issue, I was just offering what I had already done, which is not meant to be the complete solution.

That's cool, and thanks for sharing! There's no particular rush to continue working on my comments. We can leave the PR open for months. The simple fact that you shared it already makes it available to others, even if it might not be ready to be merged just yet. Many of the current plugins were shared as PRs by someone else, and then modified and committed by myself to follow the plugin architecture.

> But anyway, does that really make sense? How would spherically arranged slides be orientated? It might cause spacial disorientation in the beholder ;-)

Doing 3D presentations is certainly way more challenging than 2D, but impress.js does make that possible, so we should try to create positioning plugins that operate in 3D space. I think a generalization of what you have done here would simply be that spherical coordinates affect (x,y,z) coordinates, but don't affect rotation at all. If the author wants to have the slides facing some way other than the default "face toward z axis" then they would use the data-rotate-x/y/z attributes to achieve that. For example, it could be cool to create a presentation that is a football = a sphere with the classic white and black patches :-)

Reply:
OK, in this case it might be useful to have an (optional) `data-polar-normal` flag or something like that, which will automatically rotate the slide to be normal to the direction of (dis)placement.
Alternatively it should at least map the following `data-rotate` directions accordingly (so that z is in the direction of  the radius and x always parallel to the former x-z-plane). 
Have to poke old Euclid into action for that...